core/src/main/scala/akka/persistence/r2dbc/state/scaladsl/R2dbcDurableStateStore.scala

```diff
@@ -65,20 +66,22 @@ class R2dbcDurableStateStore[A](system: ExtendedActorSystem, config: Config, cfg
   private val log = LoggerFactory.getLogger(getClass)
   private val sharedConfigPath = cfgPath.replaceAll("""\.state$""", "")
   private val settings = R2dbcSettings(system.settings.config.getConfig(sharedConfigPath))
-  private val journalSettings = R2dbcSettings(system.settings.config.getConfig(sharedConfigPath))
   log.debug("R2DBC journal starting up with dialect [{}]", settings.dialectName)
 
   private val typedSystem = system.toTyped
   private val serialization = SerializationExtension(system)
   private val persistenceExt = Persistence(system)
-  private val stateDao = settings.connectionFactorySettings.dialect.createDurableStateDao(
-    settings,
-    ConnectionFactoryProvider(typedSystem)
-      .connectionFactoryFor(sharedConfigPath + ".connection-factory"))(typedSystem)
+  // FIXME maybe this is using the wrong executionContext, H2Dialect is using another dispatcher?
```

Comment:
this might be a regression, that would be important to follow up on before making a real release

Reply:
Not sure if this comment is related to that, but, as far as I remember H2 needed to use a blocking dispatcher because it synchronizes and filesystem stuff on the calling thread.

Reply:
yes, that is what I saw, and I think this change makes it use the default

Reply:
fixed in https://github.com/akka/akka-persistence-r2dbc/pull/518